packages/functions/src/jobs/SendConfirmation.test.ts

```diff
@@ -0,0 +1,132 @@
+import * as admin from "firebase-admin";
+import * as Functions from "../index";
+import {
+  AppointmentStatus,
+  BookingChange,
+  FunctionsApi,
+  Hospital,
+} from "@zm-blood-components/common/src";
+import { DbAppointment, DbDonor } from "../function-types";
+import {
+  deleteAppointmentsByIds,
+  getAppointmentsByIds,
+  setAppointment,
+} from "../dal/AppointmentDataAccessLayer";
+import { deleteDonor, setDonor } from "../dal/DonorDataAccessLayer";
+import { sampleUser } from "../testUtils/TestSamples";
+import { mocked } from "ts-jest/utils";
+import {
+  NotificationToDonor,
+  sendEmailToDonor,
+} from "../notifications/NotificationSender";
+import { SendConfirmationReminders } from "./CronJobs";
+import firebaseFunctionsTest from "../testUtils/FirebaseTestUtils";
+
+const wrapped = firebaseFunctionsTest.wrap(
+  Functions[FunctionsApi.BookAppointmentFunctionName]
+);
+
+jest.mock("../notifications/NotificationSender");
+const mockedNotifier = mocked(sendEmailToDonor);
+
+const DONORS = ["BookedDonorId", "AvailableDonorId", "OutOfRangeDonorId"];
+const APPOINTMENTS = [
+  "BookedAppointment",
+  "AvailableAppointment",
+  "OutOfRangeAppointment",
+];
+
+const reset = async () => {
+  for (let i = 0; i < DONORS.length; i++) {
+    await deleteDonor(DONORS[i]);
+  }
+  await deleteAppointmentsByIds(APPOINTMENTS);
+};
+
+beforeAll(async () => {
+  await reset();
+});
+
+afterEach(async () => {
+  await reset();
+});
+
+test("run on all appointments", async () => {
+  await createDonor(DONORS[0]);
+  await createDonor(DONORS[1]);
+  await createDonor(DONORS[2]);
+  const lastHour = new Date();
+  lastHour.setHours(lastHour.getHours() - 1);
+  await saveAppointment(
+    DONORS[0],
+    APPOINTMENTS[0],
+    AppointmentStatus.BOOKED,
+    admin.firestore.Timestamp.fromDate(lastHour)
+  );
+
+  await saveAppointment(
+    DONORS[1],
+    APPOINTMENTS[1],
+    AppointmentStatus.AVAILABLE,
+    admin.firestore.Timestamp.fromDate(lastHour)
+  );
+  const yesterday = new Date();
```

Comment:
Continuing mu comment from above, here is a nice example - 
Theoretically if this test runs at 23:59:59 this could fail, as `lastHour` and `yesterday` would be on the same day as they are calculated 1s after the other